shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -264,7 +266,13 @@ public FlutterEngine(
       @Nullable String[] dartVmArgs,
       boolean automaticallyRegisterPlugins,
       boolean waitForRestorationData) {
-    this.dartExecutor = new DartExecutor(flutterJNI, context.getAssets());
+    AssetManager assetManager;
+    try {
+      assetManager = context.createPackageContext(context.getPackageName(), 0).getAssets();
+    } catch (NameNotFoundException e) {
+      assetManager = context.getAssets();
+    }
+    this.dartExecutor = new DartExecutor(flutterJNI, assetManager);
```

Comment:
I think this could be inside a `finally` block, as `DartExecutor` is always initialized with the `assetManager` value, no matter there's an exception.
```suggestion
    } finally {
      this.dartExecutor = new DartExecutor(flutterJNI, assetManager);
    }
```